### PR TITLE
[wasm] Clean emsdk setup

### LINKED ISSUE #105439: [wasm] move emsdk setup from runtime to emsdk tooling
Right now src/mono/mono.proj has complicated logic to use the emsdk packages to build runtime, that setup logic should be moved to emsdk and used where do similar things (like dotnet/icu)

see https://github.com/dotnet/runtime/pull/105437 for an example

- [ ] emscripten config file
- [ ] Linux/Windows scripts `emsdk_env*`

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/105439.

This PR is enhanced version of previous attempt of cleanup done in 2 PRs: in emsdk: https://github.com/dotnet/emsdk/pull/851 and runtime: https://github.com/dotnet/runtime/pull/105612. What it does:
- removes overwriting the `emsdk_env*` script
- removes overwriting emscripten config file
- skips moving node and python nugets to `EMSDK_PATH` location
- instead it makes sure the required paths are set before provisioning emscripten - running `pre_emsdk_env` (better names are welcome)

Justification of changes in  `EmSdkRepo.Defaults.props`:
- It does not break lib tests/WBT/samples and we expect node and python nuget packages to be in nuget directory.
- Moving the changes to `WasmApp.InTree.props` was not successful - the `.props` file is imported too late to matter in the stage of checking if node / python paths are set.
- Moving the changes to `WasmApp.InTree.targets` would require a workaround that would make the target run before `_SetupEmscripten` task and also before `EmSdkRepo.Defaults.props` import that is "caching" an error about python path on Windows. Such a workaround would not be as straightforward as setting the paths in `EmSdkRepo.Defaults.props`.

### CODE REVIEW

**Comment by ilonatommy** [comment]
I have a problem with native build on Windows.

`build-native.proj` does not have access to any of properties set in:
- `src\mono\Directory.Build.props` (`ProvisionEmscriptenDir`, `EMSDK_PATH`)
- nor in `src\mono\mono.proj` (`_PreEmsdkEnvScriptPath`),

so when it executes 
`"C:\Users\user\source\repos\runtime-fork\src\native\libs\build-native.cmd" wasm Release outconfig net9.0-browser-Release-wasm -os browser icudir "C:\Users\user\source\repos\nugets\microsoft.netcore.runtime.icu.transport\9.0.0-rc.1.24373.1/runtimes/browser-wasm/native"`,
it does it with empty env variables. In other words, `.emscripten` config reads the following:
`DOTNET_EMSCRIPTEN_LLVM_ROOT`, `DOTNET_EMSCRIPTEN_NODE_JS`, `DOTNET_EMSCRIPTEN_BINARYEN_ROOT` as empty.

I have to find a way to append required variables to `_BuildNativeEnvironmentVariables` in build-native project (not from fixed paths - then it works)
```
<DOTNET_EMSCRIPTEN_LLVM_ROOT>C:\Users\user\source\repos\runtime-fork\src\mono\browser\emsdk\bin</DOTNET_EMSCRIPTEN_LLVM_ROOT>
<_BuildNativeEnvironmentVariables>$(_BuildNativeEnvironmentVariables);DOTNET_EMSCRIPTEN_LLVM_ROOT=$(DOTNET_EMSCRIPTEN_LLVM_ROOT);EM_LLVM_ROOT=$(DOTNET_EMSCRIPTEN_LLVM_ROOT)</_BuildNativeEnvironmentVariables>
```

File: src/native/libs/build-native.proj [hunk 1]
```diff
@@ -79,10 +79,14 @@
           Condition="$([MSBuild]::IsOsPlatform(Windows))">
     <PropertyGroup>
         <_BuildNativeArgs Condition="'$(Ninja)' == 'false'">$(_BuildNativeArgs) msbuild</_BuildNativeArgs>
+        <!-- how to change pass EMSDK_PATH to here? Now it's empty  -->
+        <DOTNET_EMSCRIPTEN_LLVM_ROOT>$(EMSDK_PATH)\bin</DOTNET_EMSCRIPTEN_LLVM_ROOT>
```
Comment (ilonatommy) [inline src/native/libs/build-native.proj:83]: ToDo: pass `EMSDK_PATH` to this place.

File: src/native/libs/build-native.proj [hunk 2]
```diff
@@ -78,11 +78,15 @@
...
+      <DOTNET_EMSCRIPTEN_LLVM_ROOT>$(EMSDK_PATH)\bin</DOTNET_EMSCRIPTEN_LLVM_ROOT>
+      <DOTNET_EMSCRIPTEN_NODE_JS>$(NuGetPackageRoot)runtime.$(_portableHostOS)-$(BuildArchitecture).microsoft.netcore.runtime.wasm.node.transport/$(NodePackageVersion)/tools/$(_portableHostOS)-$(BuildArchitecture)/bin/node</DOTNET_EMSCRIPTEN_NODE_JS>
+      <_BuildNativeEnvironmentVariables>$(_BuildNativeEnvironmentVariables);DOTNET_EMSCRIPTEN_LLVM_ROOT=$(DOTNET_EMSCRIPTEN_LLVM_ROOT);EM_LLVM_ROOT=$(DOTNET_EMSCRIPTEN_LLVM_ROOT);DOTNET_EMSCRIPTEN_NODE_JS=$(DOTNET_EMSCRIPTEN_NODE_JS);DOTNET_EMSCRIPTEN_BINARYEN_ROOT=$(EMSDK_PATH)</_BuildNativeEnvironmentVariables>
     </PropertyGroup>
     <!-- Run script that uses CMake to generate and build the native files. -->
     <Message Text="&quot;$(MSBuildThisFileDirectory)build-native.cmd&quot; $(_BuildNativeArgs)" Importance="High"/>
-    <Exec Command="&quot;$(MSBuildThisFileDirectory)build-native.cmd&quot; $(_BuildNativeArgs)" EnvironmentVariables="$(_BuildNativeEnvironmentVariables)" />
+    <Message Text="_BuildNativeEnvironmentVariables = $(_BuildNativeEnvironmentVariables)" Importance="High"/>
```
Comment (ilonatommy) [inline src/native/libs/build-native.proj:88]: This prints:
```
  _BuildNativeEnvironmentVariables = ;DOTNET_EMSCRIPTEN_LLVM_ROOT=\bin;EM_LLVM_ROOT=\bin;DOTNET_EMSCRIPTEN_NODE_JS=D:\a\_work\1\s\.packages\runtime.win-x64.microsoft.netcore.runtime.wasm.node.transport/9.0.0-alpha.1.24175.1/tools/win-x64/bin/node;DOTNET_EMSCRIPTEN_BINARYEN_ROOT=
```
@radekdoulik, do you have an idea how to make `EMSDK_PATH` not empty during the native build?

File: src/native/libs/build-native.proj [hunk 3]
```diff
@@ -78,11 +78,16 @@
...
+      <_BuildNativeEnvironmentVariables>$(_BuildNativeEnvironmentVariables);DOTNET_EMSCRIPTEN_LLVM_ROOT=$(DOTNET_EMSCRIPTEN_LLVM_ROOT);EM_LLVM_ROOT=$(DOTNET_EMSCRIPTEN_LLVM_ROOT);DOTNET_EMSCRIPTEN_NODE_JS=$(DOTNET_EMSCRIPTEN_NODE_JS);DOTNET_EMSCRIPTEN_BINARYEN_ROOT=$(EMSDK_PATH)</_BuildNativeEnvironmentVariables>
     </PropertyGroup>
     <!-- Run script that uses CMake to generate and build the native files. -->
-    <Message Text="&quot;$(MSBuildThisFileDirectory)build-native.cmd&quot; $(_BuildNativeArgs)" Importance="High"/>
-    <Exec Command="&quot;$(MSBuildThisFileDirectory)build-native.cmd&quot; $(_BuildNativeArgs)" EnvironmentVariables="$(_BuildNativeEnvironmentVariables)" />
+    <Message Text="&quot;$(MSBuildThisFileDirectory)build-native.cmd&quot; $(_BuildNativeArgs) with EnvironmentVariables=$(_BuildNativeEnvironmentVariables)" Importance="High"/>
+    <Exec Command="&quot;$(MSBuildThisFileDirectory)build-native.cmd&quot; $(_BuildNativeArgs)" EnvironmentVariables="$(_BuildNativeEnvironmentVariables)" EchoOff="false" />
```
Comment (ilonatommy) [inline src/native/libs/build-native.proj:90]: ToDo: revert `EchoOff="false"`

File: src/mono/browser/build/BrowserWasmApp.targets [hunk 4]
```diff
@@ -195,13 +195,17 @@
...
+      <_EmscriptenNodePath>$([MSBuild]::NormalizePath($(EmscriptenNodeToolsPath), 'bin', 'node$(_ExeExt)'))</_EmscriptenNodePath>
     </PropertyGroup>
 
     <!-- Environment variables required for running emsdk commands like `emcc` -->
     <ItemGroup Condition="'$(EmscriptenSdkToolsPath)' != ''">
       <EmscriptenEnvVars Include="DOTNET_EMSCRIPTEN_LLVM_ROOT=$(EmscriptenSdkToolsPath)bin" />
+      <EmscriptenEnvVars Include="EM_LLVM_ROOT=$(EmscriptenSdkToolsPath)bin" />
       <EmscriptenEnvVars Include="DOTNET_EMSCRIPTEN_BINARYEN_ROOT=$(EmscriptenSdkToolsPath)" />
-      <EmscriptenEnvVars Include="DOTNET_EMSCRIPTEN_NODE_JS=$([MSBuild]::NormalizePath($(EmscriptenNodeToolsPath), 'bin', 'node$(_ExeExt)'))" />
+      <EmscriptenEnvVars Include="EM_BINARYEN_ROOT=$(EmscriptenSdkToolsPath)" />
```
Comment (ilonatommy) [inline src/mono/browser/build/BrowserWasmApp.targets:206]: Isn't `EM_BINARYEN_ROOT` required only in Windows? Also, we should be missing `EMSDK_PYTHON` for Windows here.

File: src/mono/browser/build/BrowserWasmApp.targets [hunk 5]
```diff
@@ -370,6 +374,8 @@
       <EmscriptenEnvVars Include="ENABLE_BROWSER_PROFILER=$([System.Convert]::ToInt32($(WasmProfilers.Contains('browser'))))" />
       <EmscriptenEnvVars Include="RUN_AOT_COMPILATION=1" Condition="'$(RunAOTCompilation)' == 'true'" />
       <EmscriptenEnvVars Include="RUN_AOT_COMPILATION=0" Condition="'$(RunAOTCompilation)' != 'true'" />
+      <EmscriptenEnvVars Include="EM_LLVM_ROOT=$(EmscriptenSdkToolsPath)bin" />
+      <EmscriptenEnvVars Include="EM_NODE_JS=$([MSBuild]::NormalizePath($(EmscriptenNodeToolsPath), 'bin', 'node$(_ExeExt)'))" />
```
Comment (ilonatommy) [inline src/mono/browser/build/BrowserWasmApp.targets:378]: These two are duplicates of lines 204 and 208, to be removed in the next cleanup.

File: src/mono/browser/build/WasmApp.InTree.props [hunk 6]
```diff
@@ -12,6 +12,8 @@
     <EnableTargetingPackDownload>false</EnableTargetingPackDownload>
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <EMSDK_PATH Condition="'$(EMSDK_PATH)' == '' and '$(MonoProjectRoot)' != ''">$([MSBuild]::NormalizeDirectory($(MonoProjectRoot), 'browser', 'emsdk'))</EMSDK_PATH>
+    <EmscriptenNodeToolsPath>$([MSBuild]::NormalizeDirectory( $([System.IO.Path]::Combine($(NuGetPackageRoot), 'runtime.$(_portableHostOS)-$(BuildArchitecture).microsoft.netcore.runtime.wasm.node.transport', '$(NodePackageVersion)', 'tools', '$(_portableHostOS)-$(BuildArchitecture)')) ))</EmscriptenNodeToolsPath>
```
Comment (ilonatommy) [inline src/mono/browser/build/WasmApp.InTree.props:15]: How is this not being executed?
linux:
`Specified Emscripten sdk at $(EMSDK_PATH)=/__w/1/s/src/mono/browser/emsdk/ is missing some paths: $(EmscriptenNodeToolsPath)=/__w/1/s/src/mono/browser/emsdk/node/bin/`
windows:
`Specified Emscripten sdk at $(EMSDK_PATH)=D:\a\_work\1\s\src\mono\browser\emsdk\ is missing some paths: $(EmscriptenPythonToolsPath)=D:\a\_work\1\s\src\mono\browser\emsdk\python\`

**Review by radekdoulik — APPROVED** [review]
LGTM

**Review by maraf — APPROVED** [review]

**Comment by ilonatommy** [comment]
Connected failure:
https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-106403-merge-377c58235651459b9d/WasmTestOnV8-ST-System.Runtime.Tests/1/console.9f62bb6b.log?helixlogtype=result

**Pull request closed without merging**